src/build/mdn-comments.ts

```diff
@@ -0,0 +1,106 @@
+import fs from "fs/promises";
+import { basename } from "path";
+
+const basePath = new URL(
+  "../../inputfiles/mdn/files/en-us/web/api/",
+  import.meta.url,
+);
+
+function extractSummary(markdown: string): string {
+  // Remove frontmatter (--- at the beginning)
+  markdown = markdown.replace(/^---[\s\S]+?---\n/, "");
+
+  // Normalize line breaks by collapsing consecutive newlines into a single space
+  const normalizedText = markdown
+    .split("\n")
+    .map((line) => line.trim())
+    .filter(
+      (line) =>
+        line &&
+        !line.startsWith("#") &&
+        !line.startsWith(">") &&
+        !line.startsWith("{{"),
+    )
+    .join(" ")
+    .replace(
+      /\{\{\s*(Glossary|HTMLElement|SVGAttr|SVGElement|cssxref|jsxref|HTTPHeader)\s*\(\s*["']((?:\\.|[^"\\])*?)["'].*?\)\s*\}\}/gi,
+      "$2",
+    ) // Extract first argument from multiple templates, handling escaped quotes & spaces
+    .replace(
+      /\{\{\s*domxref\s*\(\s*["']((?:\\.|[^"\\])*?)["'][^}]*\)\s*\}\}/gi,
+      "$1",
+    ) // Extract first argument from domxref, handling spaces
+    .replace(
+      /\{\{\s*(?:event|jsxref|cssref|specname)\s*\|\s*([^}]+)\s*\}\}/gi,
+      "$1",
+    ) // Handle event, jsxref, cssref, etc.
+    .replace(/\{\{\s*([^}]+)\s*\}\}/g, (_, match) => `[MISSING: ${match}]`) // Catch any remaining unhandled templates
+    .replace(/\[(.*?)\]\(.*?\)/g, "$1") // Keep link text but remove URLs
+    .replace(/\s+/g, " ") // Normalize spaces
+    .replace(/\n\s*/g, "\n") // Ensure line breaks are preserved
+    .replace(/"/g, "'")
+    .trim();
+
+  // Extract the first sentence (ending in . ! or ?)
+  const sentenceMatch = normalizedText.match(/(.*?[.!?])(?=\s|$)/);
+  if (sentenceMatch) {
+    return sentenceMatch[0]; // Return the first full sentence
+  }
+
+  return normalizedText.split(" ")[0] || ""; // Fallback: first word if no sentence found
+}
+
+async function getFolders(dirPath: URL): Promise<URL[]> {
```

Comment:
getDirectories, to match other names (it's readdir and dirPath, not readfolder and folderPath)

```suggestion
async function getDirectories(dirPath: URL): Promise<URL[]> {
```